src/main/java/net/fabricmc/loader/api/LanguageAdapter.java

```diff
@@ -16,6 +16,87 @@
 
 package net.fabricmc.loader.api;
 
+/**
+ * Creates instances of objects from custom notations.
+ *
+ * <p>It enables obtaining of other JVM languages' objects with custom instantiation logic.</p>
+ *
+ * <p>A language adapter is defined as so in {@code fabric.mod.json}:
+ * <pre><blockquote>
+ *   "languageAdapter": {
+ *     "&lt;a key&gt;": "&lt;the binary name of the language adapter class&gt;"
+ *   }
+ * </blockquote></pre>
+ * Multiple keys can be present in the {@code languageAdapter} section.</p>
+ *
+ * <p>In the declaration, the language adapter is referred by its <a href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-13.html#jls-13.1">binary name</a>,
+ * such as {@code "mypackage.MyClass$Inner"}. It must have a no-argument public constructor for the Loader to instantiate.</p>
+ *
+ * <p>The {@code default} language adapter from Fabric Loader can accepet {@code value} as follows:
+ * <ul>
+ *   <li>A fully qualified reference to a class, in <a href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-13.html#jls-13.1">
+ *   binary name</a>, such as {@code package.MyClass$Inner}, where the class has a public no-argument constructor
+ *   and {@code type} is assignable from the class.
+ *
+ *   <p>An example of an entrypoint class
+ *   <pre><blockquote>
+ *     package net.fabricmc.example;
+ *     import net.fabricmc.api.ModInitializer;
+ *     public class ExampleMod implements ModInitializer {
+ *       {@literal @}Override
+ *       public void onInitialize() {}
+ *     }
+ *   </blockquote></pre>
+ *   You would declare {@code "net.fabricmc.example.ExampleMod"}.</p>
+ *
+ *   <p>For each entrypoint reference, a new instance of the class is created.
+ * 	 If this class implements two separate entrypoints, there will be two distinct
+ * 	 instances of this class in two entrypoint containers.</p>
+ * 	 </li>
+ *
+ *   <li>A fully qualified reference to a class in binary name followed by {@code ::} and a
+ *   field name. The field must be static, and {@code type} must be assignable from
+ *   the field's class.
+ *
+ *   <p>An example of an entrypoint field
+ *   <pre><blockquote>
+ *     package net.fabricmc.example;
+ *     import net.fabricmc.api.ModInitializer;
+ *     public enum ExampleMod implements ModInitializer {
+ *       INSTANCE;
```

Comment:
I dont get everyone obession with using enums for singletons, lets not use it as an example.